art/attacks/attack.py

```diff
@@ -244,6 +244,48 @@ def poison(self, x: np.ndarray, y=Optional[np.ndarray], **kwargs) -> Tuple[np.nd
         raise NotImplementedError
 
 
+class PoisoningAttackGenerator(Attack):
+    """
+    Abstract base class for poisoning attack classes that return a transformed classifier.
+    These attacks have an additional method, `poison_estimator`, that returns the poisoned classifier.
+    """
+
+    def __init__(self, generator: Optional["GENERATOR_TYPE"]) -> None:
+        """
+        :param generator: A generator
+        """
+        super().__init__(generator)
+
+    @abc.abstractmethod
+    def poison_estimator(
+        self,
+        z_trigger: np.ndarray,
+        x_target: np.ndarray,
+        batch_size: int,
+        max_iter: int,
+        lambda_p: float,
+        verbose: int,
+        **kwargs
+    ) -> "GENERATOR_TYPE":
+        """
+        Returns a poisoned version of the generator used to initialize the attack
+        :return: A poisoned generator
+        """
+        raise NotImplementedError
+
+    def z_trigger(self):
```

Comment:
Is this a property that can be decorated with `@property`?

Reply:
yes addressed